### ISSUE TITLE
Sapling space requirements are frustrating to some players

### ISSUE THREAD

**Issue by saltyseadoggo** [CorgiTaco/BYG#372]
**Is your feature request related to a problem? Please describe.**
On the All of Fabric 3 server, I watched someone rant for half an hour or so about how they couldn't get a jacaranda sapling to grow in such places as on a slope, within 'five blocks' of a wall, and so on despite light being reportedly available to the sapling. This person reportedly wasted a lot of bonemeal and became very frustrated trying to grow the tree, and another individual chimed in that "some of (BYG's) saplings just don't grow." I personally have not messed around much with growing BYG trees from saplings to be able to comment on this myself, but their sheer frustration made me want to bring the issue to light.

**Describe the solution you'd like**
I know this isn't a very specific issue, and that the conditions that these persons tried to grow the saplings in aren't very well described, but perhaps you can take a look at the saplings and see if anything would be causing them to require unusually large amounts of space, and if anything can be done to make them require less space to avoid frustrating players. I'm guessing that this person expected the tree to require space similar to what a vanilla sized tree requires, but they also reportedly tried growing a jacaranda within five blocks of a wall using bonemeal and failed. Especially concerning is if they can't be grown on naturally occurring hillsides; if this is the case, perhaps something can be done to make saplings less picky about what's around the lowest levels of the tree?

**Describe alternatives you've considered**
Many of BYG's trees are fairly large and require a lot of space, compared to vanilla trees. As such, this issue could just be Works as Intended. However, I wanted to bring this to attention simply due to how frustrated this person was trying to get your trees to grow, as a pleasant user experience I'm sure is something you'd like for BYG.

paulevs' Better End and Better Nether use smaller versions of their absolutely massive trees for when they are grown via saplings. While these smaller versions are still enormous, this gives me an idea: perhaps saplings for larger BYG trees could grow smaller trees if there isn't enough space for larger ones, so players can still grow the tree and don't become frustrated? This would also reward clearing out more space with a higher return of wood.

In summary, the solution I would suggest is
- See if you can replicate the issue of having trouble growing BYG trees on naturally generating slopes, and, if you can, see if perhaps saplings can be made more tolerant of blocks around the lower y levels of the prospective tree.
- See if anything is wrong with the space needed by BYG saplings, such as if they need a larger space than what they actually occupy, or if this is Works as Intended.
- Consider my idea of having saplings grow smaller trees, more in scale with vanilla oaks, for example, if there isn't enough room for the large and grand ones seen in the biomes, so players can still get a result from all of their bonemeal and don't become frustrated with the high space requirements.

**Comment by dzedzz** [comment]
(1.18.2 - Fabric)
I can second this. Folks on my server cannot get BYG saplings to grow at all.
Enchanted Tree saplings given an arbitrarily large flat area of grass (in one case, 20 open blocks in every direction), and fed a full stack of bonemeal, would not grow.

It's unclear as to why - working as intended, or bug. 
Do BYG saplings each require a highly specific dirt type underneath them? Can they only be manually grown within their specific biome?
Some sort of accessible list for each sapling's specific growth requirements would be extremely useful.

**Comment by Amurkura** [comment]
I've had the issue in 1.18.2 Fabric that some saplings won't grow at all in flat worlds but work fine in normal generation. (Aspen, jacaranda, pine, and zelkova). It is very frustrating and there's not a clear reason why this happens.


EDIT: I had a hunch about why this was happening: the saplings only grow if placed at y0 or above
